test/Microsoft.AspNetCore.Sockets.Common.Tests/Internal/Formatters/BinaryMessageFormatterTests.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Sockets.Internal.Formatters;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Sockets.Tests.Internal.Formatters
+{
+    public partial class BinaryMessageFormatterTests
+    {
+        [Fact]
+        public void WriteMultipleMessages()
+        {
+            var expectedEncoding = new byte[]
+            {
+                /* length: */ 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                    /* type: */ 0x01, // Binary
+                    /* body: <empty> */
+                /* length: */ 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x0E,
+                    /* type: */ 0x00, // Text
+                    /* body: */ 0x48, 0x65, 0x6C, 0x6C, 0x6F, 0x2C, 0x0D, 0x0A, 0x57, 0x6F, 0x72, 0x6C, 0x64, 0x21,
+                /* length: */ 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01,
+                    /* type: */ 0x03, // Close
+                    /* body: */ 0x41,
```

Comment:
Correct. Though the spec currently says that close is body-less. I think having a body for messages, error codes, etc. would be useful so I'll update the spec. If we need to have a more structured error/close frame body, we can do it in text form.